Xamarin.Forms.Xaml/XamlServiceProvider.cs

```diff
@@ -91,19 +92,63 @@ public void Add(Type type, object service)
 
 	class XamlValueTargetProvider : IProvideParentValues, IProvideValueTarget
 	{
+		
+		Lazy<object> targetProperty;
 		public XamlValueTargetProvider(object targetObject, INode node, HydratationContext context, object targetProperty)
```

Comment:
>the targetProperty should come from the caller, not induced by this method. 

I don't put out constructor, because i don't know all Xamarin framework and I don't know the whole part where the constructor XamlValueTargetProvider is called.
>Also, this code ignore the parameter value.

It 's true in part, targetProperty is ignored only when it is null. Then see the line 106.